src/components/FileEditorContainer.tsx

```diff
@@ -33,13 +36,33 @@ const FileEditorContainer: React.FC<FileEditorContainerProps> = () => {
     setNavigationHistory,
   } = useFileByFilepath();
 
+  const { files, flattenedFiles, expandedDirectories, handleDirectoryToggle } =
+    useFileInfoTree(filePath);
+
   const toggleChatbot = () => {
     setShowChatbot(!showChatbot);
   };
   const toggleSimilarFiles = () => {
     setShowSimilarFiles(!showSimilarFiles);
   };
 
+  useEffect(() => {
+    const handleClick = async (event: MouseEvent) => {
+      const target = event.target as HTMLElement;
+      if (target.getAttribute("data-backlink") === "true") {
+        event.preventDefault();
+        const backlinkPath = target.textContent;
+        if (backlinkPath) await openRelativePath(backlinkPath);
+      }
+    };
+
+    document.addEventListener("click", handleClick);
+
+    return () => {
+      document.removeEventListener("click", handleClick);
+    };
+  }, [openRelativePath]);
+
```

Comment:
Might not be possible depending on how the onClick render works with the HTML editor, but could we directly attach onClick listener to the backlink after the relative path is correctly set/selected? So we don't have to set this overarching useEffect that overrides the click handler.

Splitting this code logic is going to make it slightly harder to find where does the code for navigation live.

Reply:
Fixed! Originally thought it may be difficult to do given that passing in the openfile to tiptap on init would not let it access the state the openfile function needs. Modified the code to ```useRef``` and it now works without document listener